gradle/hacks/global-exclude-dependencies.gradle

```diff
@@ -0,0 +1,34 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// Globally exclude known bad dependencies to ensure they aren't included in any transitive dependency.
+
+allprojects { prj ->
+  plugins.withType(JavaPlugin) {
+    configurations.all {
```

Comment:
Don't exclude it from all configurations. Exclude these from only those configurations that are created by the java plugin. I've talked about it a few times - you have no control over what configurations are defined by plugins; if you exclude arbitrary dependencies from these, you risk hours spending debugging why something in the build doesn't work as expected.

Reply:
I kind of wonder if it's worth the risk.  If the outcome is a ClassNotFoundException blahBlah.animalSnipper.Silliness then I think we'd find the issue quickly.

Reply:
I'm not enough into how Gradle's build-time classpath (plugins etc) is isolated from the compile-time classpath that ends up in our binary distribution. The goal of these exclusions is to stop certain jars from ending up in our distro, so I'm +1 to anything that achieves that goal -- globally, without having to duplicate the exclusions everywhere.

Any updates needed in forbidden-apis?

Reply:
Correct - don't use .all, use explicit names of configurations which are actually used to declare external dependencies (most of them will come from gradle's java plugin... you can define your own but typically this won't be the case).

Reply:
> I kind of wonder if it's worth the risk. If the outcome is a ClassNotFoundException blahBlah.animalSniffer.Silliness then I think we'd find the issue quickly.

The outcome is unpredictable. Depends what the configuration you're excluding stuff from is used for. Try afterEvaluate {configurations.all { println it } } and you'll see how many there are in each project.

Reply:
Ok I see what Dawid meant here. I fixed this up in ca16326ce3188915fef685b0cec57b4ce7562655